### PR TITLE
Add Nesterov's Accelerated Gradient.

### PR DESCRIPTION
**Pull request by jnishi**
I implemented Nesterov's Accelerated Gradient(NAG).
I use NAG's formulation of http://arxiv.org/abs/1212.0901 .

### CODE REVIEW

File: chainer/optimizers/nesterov_ag.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        v -= self.lr * grad
+        param += self.momentum * self.momentum * v
+        param -= (1 + self.momentum) * self.lr * grad
+
+    def update_one_gpu(self, param, grad, v):
+        cuda.elementwise(
+            'T grad, T lr, T momentum',
+            'T param, T v',
+            '''v = v * momentum - lr * grad;
+               param += momentum * momentum * v;
```
Comment (okuta) [inline chainer/optimizers/nesterov_ag.py]: `param` is reference for device variable.
Therefor, reading and writing is very slow.
`param += momentum * momentum * v - (1 + momentum) * lr * grad;` is better.

**Comment by jnishi** [comment]
Reduces access to device variables. Please check.

**Comment by okuta** [comment]
LGTM except add reference to document.

**Comment by jnishi** [comment]
Added reference to documents. Please check.

**Comment by okuta** [comment]
OK.

**Pull request merged**